src/python/gem5/resources/client_api/jsonclient.py

```diff
@@ -54,21 +54,59 @@ def __init__(self, path: str):
         self.path = path
         self.resources = []
 
-        if Path(self.path).is_file():
-            self.resources = json.load(open(self.path))
-        elif not self._url_validator(self.path):
-            raise Exception(
-                f"Resources location '{self.path}' is not a valid path or URL."
-            )
-        else:
-            req = request.Request(self.path)
-            try:
-                response = request.urlopen(req)
-            except URLError as e:
-                raise Exception(
-                    f"Unable to open Resources location '{self.path}': {e}"
+        try:
+            # Check if it's a valid file path
+            if Path(self.path).is_file():
+                with open(self.path, encoding="utf-8") as f:
+                    self.resources = json.load(f)
+
+            # Check if it's a valid URL
+            elif not self._url_validator(self.path):
+                raise ValueError(
+                    f"'{self.path}' is not a valid file path or URL."
                 )
-            self.resources = json.loads(response.read().decode("utf-8"))
+
+            # Handle 'file://' URIs
+            elif self.path.startswith("file://"):
```

Comment:
Please use `urllib.parse.urlparse` here. Don't write your own code when there's code that exists in libraries to do what you're doing.